armadillo/src/main/java/org/molgenis/armadillo/settings/ArmadilloSettings.java

```diff
@@ -0,0 +1,14 @@
+package org.molgenis.armadillo.settings;
+
+import com.google.auto.value.AutoValue;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+@AutoValue
+public abstract class ArmadilloSettings {
+  public abstract Map<String, UserDetails> getUsers();
```

Comment:
We need a concurrent map here because multiple threads can read/write to the settings.

Reply:
Good point. I added some synchronized on the file writing bits, don't know if that helps in reality.
In practice this is a single user app so we shouldn't worry so much, but will make this fix of course :-)